python/lsst/sims/catalogs/generation/db/CompoundCatalogDBObject.py

```diff
@@ -0,0 +1,185 @@
+
+__all__ = ["CompoundCatalogDBObject"]
+
+class CompoundCatalogDBObject(CatalogDBObject):
+    """
+    This is a class for taking several CatalogDBObjects that are querying
+    the same table of the same database for the same rows (but different columns; note
+    that the columns can be transformed by the CatalogDBObjects' self.columns member),
+    and combining their queries into one.
+
+    You feed the constructor a list of CatalogDBObjects.  The CompoundCatalogDBObject
+    verifies that they all do, indeed, query the same table of the same database.  It
+    then constructs its own self.columns member (note that CompoundCatalogDBObject is
+    a daughter class of CatalogDBObject) which combines all of the requested data.
+
+    When you call query_columns, a recarray will be returned as in a CatalogDBObject.
+    Note, however, that the names of the columns of the recarray will be modified.
+    If the first CatalogDBObject in the list of CatalogDBObjects passed to the constructor
+    asks for a column named 'col1', that will be mapped to 'catName_col1' where 'catName'
+    is the CatalogDBObject's objid member.  'col2' will be mapped to 'catName_col2', etc.
+    In cases where the CatalogDBObject does not change the name of the column, the column
+    will also be returned by its original, un-mangled name.
+
+    In cases where a custom query_columns method must be implemented, this class
+    can be sub-classed and the custom method added as a member method.  In that
+    case, the _table_restriction member variable should be set to a list of table
+    names corresponding to the tables for which this class was designed.  An
+    exception will be raised if the user tries to use the CompoundCatalogDBObject
+    class to query tables for which it was not written.  _table_restriction defaults
+    to None, which means that the class is for use with any table.
+    """
+
+    # This member variable is an optional list of tables supported
+    # by a specific CompoundCatalogDBObject sub-class.  If
+    # _table_restriction==None, then any table is supported
+    _table_restriction = None
+
+    def __init__(self, catalogDbObjectList):
+        """
+        @param [in] catalogDbObjectList is a list of CatalogDBObjects that
+        all query the same database table
+        """
+
+        self._dbObjectList = catalogDbObjectList
+        self._validate_input()
+
+        self._nameList = []
+        for ix in range(len(self._dbObjectList)):
+            self._nameList.append(self._dbObjectList[ix].objid)
+
+        self._make_columns()
+        self._make_dbTypeMap()
+        self._make_dbDefaultValues()
+
+        dbo = self._dbObjectList[0]
+        self.tableid = dbo.tableid
+        self.idColKey = dbo.idColKey
+        self.raColName = dbo.raColName
+        self.decColName = dbo.decColName
+        super(CompoundCatalogDBObject, self).__init__(database=dbo.database, driver=dbo.driver,
```

Comment:
Just for my own education, why is super used here? I've noticed it in catalogDBObject as well.